src/hooks/useReportUppyInstance.js

```diff
@@ -15,51 +33,75 @@ export default function useReportUppyInstance(reportType) {
-
-    uppyInstance.use(Tus, {
-      endpoint: `${__houston_url__}/api/v1/tus`,
-      headers: {
-        'x-tus-transaction-id': assetSubmissionId,
-      },
-      removeFingerprintOnSuccess: true,
-    });
-
-    uppyInstance.on('upload', () => setUploadInProgress(true));
-
-    uppyInstance.on('complete', uppyState => {
-      const uploadObjects = get(uppyState, 'successful', []);
-      const assetReferences = uploadObjects.map(o => ({
-        path: o.name,
-        transactionId: assetSubmissionId,
-      }));
-
-      setUploadInProgress(false);
-      setFiles([...fileRef.current, ...assetReferences]);
-    });
-
-    uppyInstance.on('file-removed', (file, reason) => {
-      if (reason === 'removed-by-user') {
-        const newFiles = fileRef.current.filter(
-          f => f.path !== file.name,
-        );
-        setFiles(newFiles);
-      }
-    });
-
-    setUppy(uppyInstance);
-
-    return () => {
-      if (uppyInstance) uppyInstance.close();
-    };
-  }, []);
+  useEffect(
+    () => {
+      const uppyInstance = new Uppy({
+        meta: { type: reportType },
+        restrictions: {
+          allowedFileTypes: ['.jpg', '.jpeg', '.png'],
+        },
+        autoProceed: true,
+      });
+
+      uppyInstance.use(Tus, {
+        endpoint: `${__houston_url__}/api/v1/tus`,
+        headers: {
+          'x-tus-transaction-id': assetSubmissionId,
+        },
+        removeFingerprintOnSuccess: true,
+      });
+
+      uppyInstance.on('upload', () => setUploadInProgress(true));
+
+      uppyInstance.on('complete', uppyState => {
+        const uploadObjects = get(uppyState, 'successful', []);
+        const assetReferences = uploadObjects.map(o => ({
+          path: o.name,
+          transactionId: assetSubmissionId,
+        }));
+
+        setUploadInProgress(false);
+        setFiles([...fileRef.current, ...assetReferences]);
+      });
+
+      uppyInstance.on('file-removed', async (file, reason) => {
+        if (reason === 'removed-by-user') {
+          const uploadUrl = file?.response?.uploadURL;
+          try {
+            const fileGuid = parseFileGuidFromUploadUrl(uploadUrl);
+            await axios.delete(
+              `${__houston_url__}/api/v1/tus/${fileGuid}`,
+              {
+                headers: {
+                  'x-tus-transaction-id': assetSubmissionId,
+                },
+              },
+            );
+          } catch (error) {
+            let errorMessage = `${error.name || 'Error'} deleting ${
+              file.name
+            }.`;
+
+            if (error.message) errorMessage += ` ${error.message}`;
+
+            console.error(errorMessage);
+          }
+
+          const newFiles = fileRef.current.filter(
+            f => f.path !== file.name,
+          );
+          setFiles(newFiles);
+        }
+      });
+
+      setUppy(uppyInstance);
+
+      return () => {
+        if (uppyInstance) uppyInstance.close();
+      };
+    },
+    [reportType, assetSubmissionId],
```

Comment:
Why would 
const assetSubmissionId = useMemo(uuid, []);
ever change after import of uuid from 'uuid'?

In other words, I don't understand why this would have to be a dependency?

Actually, same with reportType. If this is an argument of the hook itself, why would this one ever change within the hook call?